### ISSUE TITLE
Component Headless run hangs during running of second spec

### ISSUE THREAD

**Issue by jordaninstil** [cypress-io/cypress#16288]
### Current behavior
Similar to this issue (https://github.com/cypress-io/cypress/issues/5475), the component test headless hangs during the running of the second spec.

As discussed in the previous issue, most likely an issue with the electron browser, as it works fine with other browsers.

### Versions

Cypress:        7.0, 7.10, 7.2.0
Browser:        Electron 89                                                                    │
Node Version:   v14.16.1
OS: macOS Big Sur

**Comment by jennifer-shehane** [comment]
@jordaninstil Could you provide the test code to reproduce this? Is your test going back in history like the other referenced issue? 

It is likely related to the tests you are running so we'll need that to narrow down the issue. Thanks!

**Comment by jordaninstil** [comment]
@jennifer-shehane Yes, the production code deals with going back in history, similar to the other issue. 

The tests work fine for all other browsers, but I will try to get some examples to you soon.

**Comment by jordaninstil** [comment]
@jennifer-shehane I've worked out the issue I was having.

There were a few tests with setup outside of a describe block

```
beforeEach(() => {
  mountComponent();
});

describe("on component rendered", () => {
    ...
```

The above caused issues when running with electron headlessly, but not with other browsers. The tests that it hung on also had interaction with the react-dom-router history, which was a coincidence and caused confusion. 

Doing the following fixed the issue

```
describe("on component rendered", () => {
  beforeEach(() => {
    mountComponent();
  });

  ...
```

Not sure if this is intentional, but apologies for the mix up.

**Comment by jennifer-shehane** [comment]
Yah that's pretty strange it would only error in electron.
